plugins/connections/ui/src/main/java/org/pentaho/di/vfs/connections/ui/dialog/ConnectionDialog.java

```diff
@@ -296,9 +300,21 @@ private void updateConnectionType( String connectionType ) {
     wDetailsWrapperComp.pack();
     wScrolledComposite.setMinSize(
       wDetailsWrapperComp.computeSize( wConnectionTypeComp.getClientArea().width, SWT.DEFAULT ) );
+
+    //root path field (ignoring for local vfs connection as the child class (LocalDetailComposite.java) would provide root path field)
+    if ( !connectionType.equals( BaseMessages.getString( PKG, "ConnectionDialog.VFSConnection.Local" ) ) ) {
+      rootPathConfig();
+    }
```

Comment:
Should not have hard-code for specific connection types.
Should instead change the Local connection composite so that it does not have the old Location field, possibly, becoming empty.
Maybe need a way for a specific connection type to not have a composite, when empty.

Reply:
Addressed